daprdocs/content/en/operations/extending/developing-grpc-components.md

```diff
@@ -0,0 +1,339 @@
+---
+type: docs
+title: "Developing gRPC-based Pluggable Components"
+linkTitle: "Developing gRPC-based Pluggable Components"
+weight: 250
+description: "Extending dapr with external gRPC-based components"
+---
+
+[gRPC-based](https://grpc.io/) Dapr components are typically run as containers or processes that communicate with the dapr main process via [Unix Domain Sockets](https://en.wikipedia.org/wiki/Unix_domain_socket).
+
+## Developing a Pluggable Component from scratch
+
+{{< tabs ".NET" >}}
+{{% codetab %}}
+
+### Step 1: Prerequisites
+
+For this tutorial, you must have minimal knowledge of [gRPC and protocol buffers](https://grpc.io/), and that you chose a programming language [that supports gRPC](https://grpc.io/docs/languages/).
+
+For simplicity, all code samples uses the generated code from the [protoc](https://developers.google.com/protocol-buffers/docs/downloads) tool or supported gRPC build tools by languages.
+
+As previously mentioned, dapr uses a Unix Domain Socket to communicate with gRPC-based components, which means that as a prerequisite you also need a UNIX-like system, or for Windows users, [WSL](https://learn.microsoft.com/en-us/windows/wsl/install) should be sufficient.
+
+Also, install [.NET Core 6+](https://dotnet.microsoft.com/en-us/download)
+
+This tutorial is based on the [official Microsoft documentation](https://learn.microsoft.com/en-us/aspnet/core/grpc/basics?view=aspnetcore-6.0#c-tooling-support-for-proto-files).
+
+### Step 2: Downloading the base template project
+
+We've prepared a blueprint project that helps you to skip a few manual steps like cloning dapr repository and adding the required libraries. Get started by downloading and unzipping it using the following <a href="/code/dotnet-memstore.zip">link</a>.
+
+This template contains all the required changes to start developing a gRPC-based component from scratch, including protobuf definitions and an unimplemented InMemory StateStore that you are going to implement.
+
+### Step 3: Building and running
+
+At this point, you are ready to build and run the component. Let's test it out by running `dotnet build`
+
+```
+➜  DaprMemStoreComponent dotnet build
```

Comment:
Nit: Not sure on windows-world but shell prompts are usually referenced by `$` instead of `➜`.

Reply:
yup, its my zsh prompt theme